### ISSUE TITLE
[🐛 Bug]: ERROR: safaridriver could not launch because it is not configured correctly or you need to authenticate. Re-run safaridriver(1) and pass the '--enable' flag to configure and/or authenticate.

### ISSUE THREAD

**Issue by fdeamicis** [SeleniumHQ/selenium#11381]
### What happened?

I'm using a MacBook with macOS Monterey Version 12.4
I'm using Eclipse with Selenium WebDriver Java version 4.4.0
I have the selenium-safari-driver-4.4.0.jar and selenium-safari-driver-4.4.0-sources.jar installed

I'm running a basic script using SafariDriver as my browser

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.edge.EdgeDriver;
import org.openqa.selenium.safari.SafariDriver;

public class ScriptSafari {

	public static void main(String[] args) throws InterruptedException {
		

		WebDriver driver=new SafariDriver();

		driver.navigate().to("https://www.google.com");
		
		driver.close();
	}

}

The Develop Menu of the Safari browser has Allow Remote Automation checked
I've tried running sudo safaridriver --enable
I've tried running sudo /usr/bin/safaridriver --enable
Whenever I execute my script, I keep seeing the error below.  Has this functionality broke in later versions of the MacBook Safari or is the issue with Selenium Webdriver

ERROR: safaridriver could not launch because it is not configured
correctly or you need to authenticate. Re-run safaridriver(1) and
pass the '--enable' flag to configure and/or authenticate.
For more information, consult the safaridriver(1) man page.
Exception in thread "main" org.openqa.selenium.SessionNotCreatedException: Could not start a new session. Possible causes are invalid address of the remote server or browser start-up failure.
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:587)
	at org.openqa.selenium.remote.RemoteWebDriver.startSession(RemoteWebDriver.java:264)
	at org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:179)
	at org.openqa.selenium.safari.SafariDriver.<init>(SafariDriver.java:79)
	at org.openqa.selenium.safari.SafariDriver.<init>(SafariDriver.java:58)
	at org.openqa.selenium.safari.SafariDriver.<init>(SafariDriver.java:49)
	at ScriptSafari.main(ScriptSafari.java:14)
Caused by: org.openqa.selenium.WebDriverException: Driver server process died prematurely.
Build info: version: '4.3.0', revision: 'a4995e2c09*'
System info: host: 'CQ-CE-47346-mac', ip: 'fe80:0:0:0:86a:c9dd:eddf:3e3e%en0', os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '11.4', java.version: '16.0.1'
Driver info: driver.version: SafariDriver
	at org.openqa.selenium.remote.service.DriverService.start(DriverService.java:226)
	at org.openqa.selenium.remote.service.DriverCommandExecutor.execute(DriverCommandExecutor.java:98)
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:569)
	... 6 more
<img width="587" alt="Screen Shot 2022-10-28 at 6 45 24 PM" src="https://user-images.githubusercontent.com/4235950/206333575-dfeb7fff-fa4c-4f14-b8bc-fe0147eff949.png">
<img width="533" alt="Screen Shot 2022-10-28 at 6 50 45 PM" src="https://user-images.githubusercontent.com/4235950/206333660-62725f9d-f3ee-4844-8789-2db4a15f125a.png">
<img width="572" alt="Screen Shot 2022-10-28 at 7 27 47 PM" src="https://user-images.githubusercontent.com/4235950/206333718-f51749ef-b8ed-484d-b35a-d39cb7f00abd.png">
<img width="746" alt="Screen Shot 2022-10-26 at 6 53 11 PM" src="https://user-images.githubusercontent.com/4235950/206333781-f23ee38b-9c65-4bc4-be1c-74f642ebae43.png">
<img width="443" alt="Screen Shot 2022-10-28 at 7 44 02 PM" src="https://user-images.githubusercontent.com/4235950/206333840-b77d036a-59c4-4cb1-b682-7bde5279a5ad.png">
<img width="572" alt="Screen Shot 2022-10-28 at 8 13 52 PM" src="https://user-images.githubusercontent.com/4235950/206334000-54ac4edc-b387-44e1-a727-cc0eae467048.png">
<img width="568" alt="Screen Shot 2022-10-28 at 8 20 06 PM" src="https://user-images.githubusercontent.com/4235950/206334062-fa57e53a-cdc8-4876-ba90-9c7aeebeead3.png">
<img width="1302" alt="Screen Shot 2022-10-28 at 10 46 37 PM" src="https://user-images.githubusercontent.com/4235950/206334109-be95eaf9-9495-483c-96d4-0991e10ec374.png">

### How can we reproduce the issue?

```shell
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.edge.EdgeDriver;
import org.openqa.selenium.safari.SafariDriver;

public class ScriptSafari {

	public static void main(String[] args) throws InterruptedException {
		

		WebDriver driver=new SafariDriver();

		driver.navigate().to("https://www.google.com");
		
		driver.close();
	}

}
```

### Relevant log output

```shell
ERROR: safaridriver could not launch because it is not configured
correctly or you need to authenticate. Re-run safaridriver(1) and
pass the '--enable' flag to configure and/or authenticate.
For more information, consult the safaridriver(1) man page.
Exception in thread "main" org.openqa.selenium.SessionNotCreatedException: Could not start a new session. Possible causes are invalid address of the remote server or browser start-up failure.
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:587)
	at org.openqa.selenium.remote.RemoteWebDriver.startSession(RemoteWebDriver.java:264)
	at org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:179)
	at org.openqa.selenium.safari.SafariDriver.<init>(SafariDriver.java:79)
	at org.openqa.selenium.safari.SafariDriver.<init>(SafariDriver.java:58)
	at org.openqa.selenium.safari.SafariDriver.<init>(SafariDriver.java:49)
	at ScriptSafari.main(ScriptSafari.java:14)
Caused by: org.openqa.selenium.WebDriverException: Driver server process died prematurely.
Build info: version: '4.3.0', revision: 'a4995e2c09*'
System info: host: 'CQ-CE-47346-mac', ip: 'fe80:0:0:0:86a:c9dd:eddf:3e3e%en0', os.name: 'Mac OS X', os.arch: 'x86_64', os.version: '11.4', java.version: '16.0.1'
Driver info: driver.version: SafariDriver
	at org.openqa.selenium.remote.service.DriverService.start(DriverService.java:226)
	at org.openqa.selenium.remote.service.DriverCommandExecutor.execute(DriverCommandExecutor.java:98)
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:569)
	... 6 more
```

### Operating System

I'm using a MacBook with macOS Monterey Version 12.4

### Selenium version

I'm using Selenium WebDriver Java version 4.4.0 

### What are the browser(s) and version(s) where you see this issue?

Safari Version 15.5 (17613.2.7.1.8)

### What are the browser driver(s) and version(s) where you see this issue?

SafariDriver built in to Safari

**Comment by diemol** [comment]
I do not think you should use `sudo` to run `safaridriver --enable`. Have you tried that without `sudo`?

**Comment by fdeamicis** [comment]
I tried without sudo and I get the same problem.  I thought it was a rights/privilege issue so I used sudo to elevate my privileges.  It didn't matter.  I don't know if its an  issue with SafariDriver or if Selenium is just not connecting to SafariDriver properly

**Comment by diemol** [comment]
I suspect it is a problem in your environment, otherwise we would see tons of issues about this topic. What happens when you try on a different laptop?

**Comment by fdeamicis** [comment]
I suspect you may be correct, however, I do have admin privileges on my Macbook and I'm not sure why it would fail.  Unfortunately, I don't have another MacBook to try it on.  If this has worked successfully for others, I would love to hear from you.

**Comment by titusfortner** [comment]
Are the same limitations in place for using Technology Preview?

Download [Safari Technology Preview](https://developer.apple.com/safari/technology-preview/) and try:
```
        SafariOptions options = new SafariOptions();
        options.setUseTechnologyPreview(true);
        driver = new SafariDriver(options);
```

**Comment by fdeamicis** [comment]
No dice.  Technology Preview throws the same error.  Perhaps, it is  an issue with the SafariDriver implementation on Monterey and this might not have been a problem with older operating systems.  I can't prove it though

**Comment by fdeamicis** [comment]
It's also possible there is some configuration I'm not setting on my MacBook.

**Comment by titusfortner** [comment]
Presumably if that were the issue, there would be a lot more people complaining. Monterey has been out for over a year.

Did you try turning it off and turning it back on again? :) seriously, though, maybe a process didn't get killed properly. 

Not sure how else to help here.

**Comment by diemol** [comment]
I will close this as there is not much for us to do, seems like an issue on that particular machine.

**Comment by krmahadevan** [comment]
Today I stumbled into this problem and haven't been able to get safari working.

Here's some additional information.

I am running on Ventura (13.5.2 (22G91))

Here's what my safaridriver locations look like

```shell
$ ls -ltr /usr/bin/safaridriver
lrwxr-xr-x  1 root  wheel  42 Sep  2 13:05 /usr/bin/safaridriver -> /System/Cryptexes/App/usr/bin/safaridriver
```

And when I try to run safaridriver manually it conks

```
$ /usr/bin/safaridriver -p 8888
Unable to start the server: Operation not permitted
```
The same runs fine, when I use `sudo` 

```shell
$ sudo /usr/bin/safaridriver -p 8888
```

```shell
curl -X GET  http://localhost:8888/status
{"value":{"message":"","ready":true}}
```

Safari version: `Version 16.6 (18615.3.12.11.2)`

**Comment by krmahadevan** [comment]
Adding another update.

I noticed that there was an OSX upgrade. So I upgraded to `Ventura 13.6 (22G120)` .

The upgrade caused me to reboot my machine. After that when I tried to run `safaridriver` it went through fine without any issues. 

So am not sure if its the upgrade that fixed this issue (or) if the reboot (after running `sudo safaridriver --enable`) that fixed the issue.

Just leaving this information here, so that it can help someone who may have this issue
